crates/next-core/js/src/overlay/internal/components/Tabs/index.tsx

```diff
@@ -0,0 +1,234 @@
+// TODO: replace with `@react/tabs`
+
+import * as React from "react";
+import { noop as css } from "../../helpers/noop-template";
+import { createContext } from "../../hooks/context";
+import { clsx } from "../../helpers/clsx";
+
+type Merge<P1 = {}, P2 = {}> = Omit<P1, keyof P2> & P2;
+
+type PropsWithAs<As, OwnProps = {}> = As extends ""
+  ? { as: keyof JSX.IntrinsicElements }
+  : As extends React.JSXElementConstructor<infer P>
+  ? Merge<P, OwnProps & { as: As }>
+  : As extends keyof JSX.IntrinsicElements
+  ? Merge<JSX.IntrinsicElements[As], OwnProps & { as: As }>
+  : never;
+
+type TabRefs = Record<string, HTMLElement | undefined>;
+
+type TabsContextType = {
+  selectedId: string;
+  tabRefs: Readonly<TabRefs>;
+  registerTabRef: (id: string, el: HTMLElement | null) => void;
+  onSelectTab: (id: string) => void;
+};
+
+const [TabsProvider, useTabsContext] = createContext<TabsContextType>("Tabs");
+
+export type TabsProps = {
+  defaultId: string;
+  selectedId?: string;
+  onChange?: (id: string) => void;
+  children: React.ReactNode;
+};
+
+export function Tabs({
+  defaultId,
+  selectedId: controlledId,
+  onChange,
+  children,
+}: TabsProps) {
+  const [selectedId, setSelectedId] = React.useState(controlledId ?? defaultId);
+  const [checkId, setCheckId] = React.useState(false);
```

Comment:
nit: maybe `shouldCheckId`? At a glance this seemed like another string id like above.

Reply:
done